core/lib/types/src/commitment/mod.rs

```diff
@@ -92,6 +93,16 @@ pub struct L1BatchMetadata {
     /// commitment to the transactions in the batch.
     pub bootloader_initial_content_commitment: Option<H256>,
     pub state_diffs_compressed: Vec<u8>,
+    /// Hash of packed state diffs. It's present only for post-gateway batches.
+    pub state_diff_hash: Option<H256>,
+    /// Root hash of the local logs tree. Tree contains logs that were produced on this chain.
+    /// It's present only for post-gateway batches.
+    pub local_root: Option<H256>,
+    /// Root hash of the aggregated logs tree. Tree aggregates `local_root`s of chains that settle on this chain.
+    /// It's present only for post-gateway batches.
+    pub aggregation_root: Option<H256>,
```

Comment:
Maybe a dumb suggestion: If these fields are supposed to be present together, maybe it makes sense to extract them to a new struct?

```rust
pub struct L1BatchGatewayMetadata {
    pub state_diff_hash: H256,
    // ...
}

pub struct L1BatchMetadata {
    // ...
    pub gateway: Option<L1BatchGatewayMetadata>,
}
```

Reply:
What's the benefit? Yeah, it seems like with current implementation they will be present together but I don't think we should rely on it. Comment I left is somewhat misleading, they won't be present only for post-gateway, but they are expected to be present for each post-gateway batch.